### PR TITLE
Feature/task 4960 : maven and spring profiles

### PR DESCRIPTION
**Pull request by msamandi**
[XXXX-<Title> - Please use the Work Item number and Title as PR Name, not subtasks]

#### 📲 What

maven and spring profiles
#### 🤔 Why

Why it's needed, background context.

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: java/src/main/resources/application-azure.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+azure:
+  application-insights:
+    instrumentation-key: xxxxxx
+    enabled: false
+  keyvault:
+    enabled: false
+    uri: https://amido-stacks-tmp.vault.azure.net/
+    client-id: xxxxxx
+    client-key: xxxxxx
+    tenant-id: xxxxxx
```
Comment (sdavis-amido) [inline java/src/main/resources/application-azure.yml:10]: Needs a newline here or will fail the yaml linting

File: java/src/main/resources/application.yml [hunk 2]
```diff
@@ -1,4 +1,11 @@
 spring:
+  profiles:
+    include:
+      - "@aws.profile.name@"
+      - "@azure.profile.name@"
+      - "@cosmosdb.profile.name@"
+      - "@dynamodb.profile.name@"
```
Comment (sdavis-amido) [inline java/src/main/resources/application.yml:7]: This in incorrect and should have all of the possible profiles in it

File: java/src/main/resources/local/application-cosmodb.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+azure:
+  cosmos:
+    uri: https://localhost:8081
+    database: Stacks
+    key: ${COSMOSDB_KEY}
```
Comment (sdavis-amido) [inline java/src/main/resources/local/application-cosmodb.yml:5]: Requires a newline

File: java/src/main/resources/local/application-aws.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+aws:
+  xray:
+    enabled: ${AWS_XRAY_ENABLED:false}
+  secretsmanager:
+    enabled: ${AWS_SECRETS_ENABLED:false}
+
+# AWS Secrets Manager imports
+spring.config.import:
+  - optional:aws-secretsmanager:/stacks-secret/example-1/
+  - optional:aws-secretsmanager:/stacks-secret/example-2/
```
Comment (sdavis-amido) [inline java/src/main/resources/local/application-aws.yml:10]: Possibly requires a newline, check it

`python3 -m yamllint -sc yamllint.conf . yamllint.conf`

File: java/pom.xml [hunk 5]
```diff
@@ -378,6 +341,64 @@
   </build>
 
   <profiles>
+    <!-- START FEATURE PROFILES -->
```
Comment (sdavis-amido) [inline java/pom.xml:344]: Needs all the profiles added, not just these 4

**Review by sdavis-amido — CHANGES_REQUESTED** [review]

**Comment by sdavis-amido** [comment]
Also looks like there are Snyk issues that need addressing :-( 

image

**Comment by sdavis-amido** [comment]
image

File: java/pom.xml (hunk 5 above)
Comment (msamandi) [reply java/pom.xml:344]: as agreeed no profile will be added for messaging feature

**Pull request closed without merging**